overlord/restart/restart.go

```diff
@@ -577,8 +577,22 @@ func processRestartForChange(chg *state.Change) error {
 			logger.Noticef("cannot notify about pending reboot: %v", err)
 		}
 		logger.Noticef("Postponing restart until a manual system restart allows to continue")
-		return nil
+		return
 	}
 	Request(chg.State(), rp.RestartType, &boot.RebootInfo{RebootRequired: true, BootloaderOptions: rp.BootloaderOptions})
-	return nil
+}
+
+// MockRestartForChange is added solely for unit test purposes, to help simulate restarts.
+func MockRestartForChange(chg *state.Change) {
```

Comment:
maybe MockAfterRestartForChange is a clearer name for this?

Reply:
Yep that is indeed clearer